test/e2e-layering/onclusterbuild_test.go

```diff
@@ -0,0 +1,230 @@
+package e2e_layering
+
+import (
+	"context"
+	"flag"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/util/retry"
+
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/openshift/machine-config-operator/pkg/controller/build"
+	"github.com/openshift/machine-config-operator/test/framework"
+	"github.com/openshift/machine-config-operator/test/helpers"
+	"github.com/stretchr/testify/require"
+
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// The MachineConfigPool to create for the tests.
+	layeredMCPName string = "layered"
+
+	// The ImageStream name to use for the tests.
+	imagestreamName string = "os-image"
+
+	// The name of the global pull secret copy to use for the tests.
+	globalPullSecretCloneName string = "global-pull-secret-copy"
+
+	// The custom Dockerfile content to build for the tests.
+	cowsayDockerfile string = `FROM quay.io/centos/centos:stream9 AS centos
+RUN dnf install -y epel-release
+FROM quay.io/zzlotnik/testing:hotdogcart-containers AS hotdogcart
```

Comment:
suggestions: I'm fine with using Zack's hotdogcart here for now :D but it's best if we don't depend on external quay images for our CI runs, would we be able to easily do an alternative?

Reply:
I can remove it for now :). I was being a bit punchy when I put it in there.